src/python/pants/vcs/changed.py

```diff
@@ -40,9 +42,21 @@ class ChangedAddresses(Collection[Address]):
 
 
 @rule
-async def find_changed_owners(request: ChangedRequest) -> ChangedAddresses:
-    owners = await Get(Owners, OwnersRequest(request.sources, filter_by_global_options=True))
-    if request.dependees == DependeesOption.NONE:
+async def find_changed_owners(
+    request: ChangedRequest, specs_filter: SpecsFilter
+) -> ChangedAddresses:
+    no_dependees = request.dependees == DependeesOption.NONE
+    owners = await Get(
+        Owners,
+        OwnersRequest(
+            request.sources,
+            # If `--changed-dependees` is used, we cannot eagerly filter out root targets. We
+            # need to first find their dependees, and only then should we filter. See
+            # https://github.com/pantsbuild/pants/issues/15544
+            filter_by_global_options=no_dependees,
```

Comment:
Should this be using `bool(specs_filter)` instead, in order to ensure that regexes are applied as well?

Reply:
Regexes will be applied via the last part of this rule -- `no_dependees` only determines whether we eagerly filter with the roots, or defer the filtering until after dependees have been found